partitioner.go

```diff
@@ -82,18 +82,20 @@ func (p *roundRobinPartitioner) RequiresConsistency() bool {
 	return false
 }
 
+type hasherFunc func() hash.Hash32
```

Comment:
I wouldn't even bother naming this type, especially since it's weird for a non-exported interface to be an argument to an exported method.